### PR TITLE
[WIP] Toast counts down for 3 seconds before placing the bet

### PR DESCRIPTION
**Pull request by akrolsmir**
There's some weird timing/React hook error that sometimes allows this thing to bet twice? IDK, my brain is fried trying to understand it.

Also, progress bar that scrolls down might be cooler than a number countdown? See https://www.npmjs.com/package/react-toastify, although the API for that is more unwieldy/less Tailwind-friendly?

### CODE REVIEW

File: web/components/contract/contract-card.tsx [hunk 1]
```diff
@@ -346,3 +395,64 @@ export function NumericResolutionOrExpectation(props: {
...
+      setSecondsLeft((seconds) => seconds - 1)
+    }, 1000)
+    return () => clearInterval(interval)
+  }, [])
+
+  if (secondsLeft <= 0) {
+    console.log('finishing')
+    onToastFinish()
+    // return null
+  }
```
Comment (jahooma) [inline web/components/contract/contract-card.tsx:423]: Shouldn't this be in a useEffect?

e.g.
```
useEffect(() => {
  if (secondsLeft <= 0) {
    console.log('finishing')
    onToastFinish()
  }
}, [secondsLeft])
```

Comment (jahooma) [reply web/components/contract/contract-card.tsx:423]: Also, why not have a single setTimeout for the whole duration? You could cancel with clearTimeout on unmount.

Comment (jahooma) [reply web/components/contract/contract-card.tsx:423]: Oh I guess you render the seconds left...

**Comment by akrolsmir** [comment]
Eh, maybe will pick this up someday.

I'm not sure about the "toast" implementation vs "something changes on the card directly", which would be less intrusive/hard to track.

**Pull request closed without merging**